src/test/regress/specs/isolation_replace_wait_function.spec

```diff
@@ -0,0 +1,69 @@
+# allow isolation tester to recognize locks on workers
+
+session "s1"
+
+step "pg-96-rename"
+{
+  -- break this into a separate step because each step can be at most 1024 chars
+  ALTER FUNCTION pg_catalog.pg_blocking_pids(integer)
+    RENAME TO old_pg_blocking_pids;
+}
+
+step "pg-96-create"
+{
+  CREATE FUNCTION pg_catalog.pg_blocking_pids(pTarget integer)
+  RETURNS int4[] AS $$
+    DECLARE
+      mLocalPids int4[];
+      mRemotePids int4[];
+      mTransactionNum int8;
+    BEGIN
+      SELECT pg_catalog.old_pg_blocking_pids(pTarget) INTO mLocalPids;
+
+      IF (array_length(mLocalPids, 1) > 0) THEN
+        RETURN mLocalPids;
+      END IF;
+
+      SELECT transaction_number INTO mTransactionNum FROM get_all_active_transactions() WHERE process_id = pTarget;
+      
+      SELECT array_agg(process_id) INTO mRemotePids FROM (
+        WITH map AS (
+          SELECT process_id, transaction_number FROM get_all_active_transactions()
+        ), edge AS (
+          SELECT blocking_transaction_num FROM dump_global_wait_edges()
+          WHERE waiting_transaction_num = mTransactionNum
+        )
+        SELECT map.process_id FROM map, edge
+        WHERE map.transaction_number = edge.blocking_transaction_num
+      ) AS sub;
+
+      RETURN mRemotePids;
+    END;
+  $$ LANGUAGE plpgsql;
+}
+
+step "pg10-setup"
+{
+  ALTER FUNCTION pg_catalog.pg_isolation_test_session_is_blocked(integer, integer[])
+    RENAME TO old_pg_isolation_test_session_is_blocked;
+  
+  CREATE FUNCTION pg_catalog.pg_isolation_test_session_is_blocked(pTarget integer, pSources integer[])
+  RETURNS boolean AS $$
+    DECLARE
+      mTransactionNum int8;
+    BEGIN
+      IF pg_catalog.old_pg_isolation_test_session_is_blocked(pTarget, pSources) THEN
+        RETURN true;
+      END IF;
+
+      SELECT transaction_number INTO mTransactionNum FROM get_all_active_transactions() WHERE process_id = pTarget;
+
+      -- pg doesn't think we're blocked locally, so check whether we're blocked globally
+      RETURN EXISTS (
+        SELECT 1 FROM dump_global_wait_edges() WHERE waiting_transaction_num = mTransactionNum
+      );
```

Comment:
I also thought it would but it seems to increase run time of the entire suite by 0.1sec

Interestingly, listening to a youtube video in the background increases run time by 1.3 sec, nearly 12%!

This seems reasonable, all we need to do is open a connection to each of the local workers and run a quick c function, though it might get slower if we build complicated graphs.